core/src/test/scala/org/apache/spark/SortShuffleSuite.scala

```diff
@@ -39,12 +39,12 @@ class SortShuffleSuite extends ShuffleSuite with BeforeAndAfterAll {
   override def beforeAll() {
     super.beforeAll()
     conf.set("spark.shuffle.manager", "sort")
+    conf.set("spark.local.dir", Utils.createTempDir().getAbsolutePath)
   }
 
   override def beforeEach(): Unit = {
     super.beforeEach()
     tempDir = Utils.createTempDir()
-    conf.set("spark.local.dir", tempDir.getAbsolutePath)
```

Comment:
@heary-cao I am sorry, probably I didn't understand yet. Here it looks we create a temp directory, `tempDir`,  before a test and remove `tempDir` after a test recusively. How come a directory could leave undeleted? 

Or do you mean actually the test is failed because it fails to remove the temp directory as they have exclusive lock only on Windows?